f5_openstack_agent/lbaasv2/drivers/bigip/network_service.py

```
@@ -500,8 +534,11 @@ def get_bigip_net_short_name(self, bigip, tenant_id, network_name):
     @staticmethod
     def get_neutron_net_short_name(network):
         # Return <network_type>-<seg_id> for neutron network
-        net_type = network['provider:network_type']
-        net_seg_key = network['provider:segmentation_id']
+        net_type = network.get('provider:network_type', "")
+        if not net_type:
+            raise f5_ex.InvalidNetworkType
+        net_seg_key = network.get('provider:segmentation_id', 0)
```

We should probably raise the InvalidNetworkType exception.